packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel/method_channel_field_value.dart

```diff
@@ -23,13 +31,22 @@ enum FieldValueType {
 }
 
 /// MethodChannel implementation of a [FieldValuePlatform].
-class MethodChannelFieldValue extends FieldValuePlatform {
+class MethodChannelFieldValue {
```

Comment:
Why don't these values extend FieldValuePlatform anymore? That's the thing that ensures they have a "delegate" inside them, right?

Reply:
`MethodChannelFieldValue` doesn't have a delegate inside. It *is* the delegate (of FieldValue). Previously it had a null delegate, now it doesn't have one.

Reply:
Eeeek, you're right, now I remember that stuff. The new generic version looks much nicer, thanks!